### ISSUE TITLE
Determine minimal supported Django version

### ISSUE THREAD

**Issue by nijel** [WeblateOrg/weblate#3552]
For the 4.0 milestone minimal required Django version might be increased. Things to consider:

* Django 2.2 is currently supported LTS release, next LTS will be in 2021, see https://www.djangoproject.com/download/
* Our locales hack might be removed once https://github.com/django/django/pull/12332 is merged into Django, this might be actually an incentive to upgrade
* With Django 3.0 Weblate currently raises quite some warnings because we keep compatibility with 2.2, see https://github.com/WeblateOrg/weblate/pull/3551#issuecomment-593879883

**Comment by comradekingu** [comment]
Other than the option of faffing about with conditionals, looks like 3.0 will make its way down from Debian Experimental in time for a new release sooner or later.

**Comment by nijel** [comment]
I think Debian might want to stick with 2.2 as that is a LTS release.

**Comment by nijel** [comment]
Seems that https://github.com/django/django/pull/12332 will be backported to 2.2 as well.

**Comment by nijel** [comment]
Requiring Django 3.0 we would mean ditching Python 3.5 support. Not sure how much problematic that would be. It would allow to use some of the [Python 3.6 features](https://docs.python.org/3/whatsnew/3.6.html) such as f-strings or type annotations.
